### PR TITLE
adds subnet for Memcached cluster

### PR DESCRIPTION
**Pull request by scriptnull**
https://github.com/devops-recipes/prov_aws_vpc_terraform/issues/2

Creates a private subnet which could be used to spin up Amazon ElastiCache Memcached instances.

### CODE REVIEW

**Comment by avinci** [comment]
again why do we need subnet?

**Comment by scriptnull** [comment]
Our current VPC provisioning provides us with only one subnet, which is public and is meant for spinning up EC2 instances.

Memcached wiki guidelines suggested "not to expose memcached directly to the internet" and so I thought having a separate private subnet for memcached which would give better isolation. I also guess that most users would have a use case where they want to spin up memcached clusters in an isolated environment like this.

**Comment by avinci** [comment]
I understand, then we should call it private and not memcached.

File: shippable.yml [hunk 1]
```diff
@@ -43,6 +43,7 @@ jobs:
                 - vpc_cidr_block: "10.10.0.0/16"
                 - vpc_access_from_ip_range: "0.0.0.0/0"
                 - vpc_public_subnet_1_cidr: "10.10.10.0/24"
+                - vpc_memcached_subnet_cidr: "10.10.11.0/24"
```
Comment (avinci) [inline shippable.yml]: vpc_private_subnet_1_cidr

File: terraform.tfvars [hunk 2]
```diff
@@ -4,4 +4,5 @@ vpc_region="${vpc_region}"
 vpc_name="${vpc_name}"
 vpc_cidr_block="${vpc_cidr_block}"
 vpc_public_subnet_1_cidr="${vpc_public_subnet_1_cidr}"
+vpc_memcached_subnet_cidr="${vpc_memcached_subnet_cidr}"
```
Comment (avinci) [inline terraform.tfvars]: need to be private

**Comment by scriptnull** [comment]
Closing this in favour of https://github.com/devops-recipes/prov_aws_vpc_terraform/pull/4

**Pull request closed without merging**